### PR TITLE
add to the health check to inspect cluster health that it has a GOOD for all our instances

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5552/

**Comment by dothebart** [comment]
breaks active_failover tests - needs to be fixed.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5555/

**Comment by dothebart** [comment]
tests blue except unrelated errors.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5677

**Comment by dothebart** [comment]
tests green

File: js/client/modules/@arangodb/process-utils.js [hunk 1]
```diff
@@ -1050,6 +1050,18 @@ function checkInstanceAlive (instanceInfo, options) {
...
+  if (rc && options.cluster && instanceInfo.arangods.length > 1) {
+    try {
+      let health = require('internal').clusterHealth();
+      rc = instanceInfo.arangods.reduce((previous, arangod) => {
+        if (arangod.role === "agent") return true;
+        if (arangod.role === "single") return true;
+        return health.hasOwnProperty(arangod.id) && health[arangod.id].Status === "GOOD";
+      }, true);
+    } catch (x) {
+      print(Date() + "ClusterHealthCheck failed: " + x);
```
Comment (jsteemann) [inline js/client/modules/@arangodb/process-utils.js]:
```suggestion
      print(Date() + " ClusterHealthCheck failed: " + x);
```

File: js/client/modules/@arangodb/process-utils.js [hunk 2]
```diff
@@ -1497,6 +1509,18 @@ function checkClusterAlive(options, instanceInfo, addArgs) {
       throw new Error('cluster startup timed out after 10 minutes!');
     }
   }
+  print("Determining server IDs");
+  instanceInfo.arangods.forEach(arangod => {
+    // agents don't support the ID call...
+    if ((arangod.role !== "agent") && (arangod.role !== "single")) {
+      const reply = download(arangod.url + '/_db/_system/_admin/server/id', '', makeAuthorizationHeaders(instanceInfo.authOpts));
+      if (reply.error || reply.code !== 200) {
+        throw new Error("Server has no detecteable ID! " + JSON.stringify(reply) + "\n" + JSON.stringify(arangod));
```
Comment (jsteemann) [inline js/client/modules/@arangodb/process-utils.js]:
```suggestion
        throw new Error("Server has no detectable ID! " + JSON.stringify(reply) + "\n" + JSON.stringify(arangod));
```

**Pull request merged**